packages/cli/src/commands/MigrationCommandFactory.ts

```
@@ -164,6 +168,16 @@ export class MigrationCommandFactory {
     CLIHelper.dump(colors.green(`${ret.fileName} successfully created`));
   }
 
+  private static async handleCheckCommand(migrator: IMigrator, orm: MikroORM): Promise<void> {
+    const diff = await migrator.checkMigrationNeeded();
+
+    if (diff.up.length === 0) {
+      return CLIHelper.dump(colors.green(`No changes required, schema is up-to-date`));
+    }
+    await orm.close(true);
+    throw new Error(`Changes detected. Please create migrations to update schema.`);
```

i am not entirely sure i like the idea of throwing here. maybe `process.exit(1)` would be better

---

That's funny, because originally I was going to use `process.exit(1)` but thought it might be better to throw an error lol

---

i dont like the idea of showing a stack trace for this, sounds quite confusing. and for automation you only care about the exit code